MAVProxy/modules/mavproxy_misseditor/missionEditorFrame.py

```diff
@@ -941,6 +948,26 @@ def set_grad_dist(self):
             grad = format(grad, '.1f')
             self.grid_mission.SetCellValue(row, ME_DIST_COL, str(dist))
             self.grid_mission.SetCellValue(row, ME_ANGLE_COL, str(grad))
+            
+    def set_agl(self):
+        '''update agl altitude when changing cell values'''
+        home_def_alt = float(self.label_home_alt_value.GetLabel())
+        numrows = self.grid_mission.GetNumberRows()
+        for row in range(0,numrows):
+            command = self.grid_mission.GetCellValue(row, ME_COMMAND_COL)
+            lat = float(self.grid_mission.GetCellValue(row, ME_LAT_COL))
+            lon = float(self.grid_mission.GetCellValue(row, ME_LON_COL))
+            agl = 0.0
+            elevation = self.ElevationModel.GetElevation(lat, lon)
```

Comment:
need to check for None here, eg terrain unavailable

Reply:
Done.